risc0/zkvm/src/host/api/server.rs

```diff
@@ -835,12 +833,27 @@ fn execute_redis(
     exec: &mut ExecutorImpl,
     params: super::RedisParams,
 ) -> Result<crate::Session> {
-    let client = redis::Client::open(params.url)?;
-    let mut connection = client.get_connection()?;
-    exec.run_with_callback(|segment| {
+    let (sender, receiver) = kanal::unbounded();
+    let join_handle: std::thread::JoinHandle<()> = std::thread::spawn(move || {
+        let client = redis::Client::open(params.url)
+            .map_err(anyhow::Error::new)
+            .unwrap();
+        let mut connection = client.get_connection().map_err(anyhow::Error::new).unwrap();
+        while let Ok((segment_key, segment_bytes)) = receiver.recv() {
+            redis::cmd("SETEX")
+                .arg(segment_key)
+                .arg(params.ttl)
+                .arg(segment_bytes)
+                .exec(&mut connection)
+                .map_err(anyhow::Error::new)
+                .unwrap();
+        }
+    });
+
+    let session = exec.run_with_callback(|segment| {
         let segment_bytes = bincode::serialize(&segment)?;
         let segment_key = format!("{}:{}", params.key, segment.index);
-        connection.set_ex(segment_key.clone(), segment_bytes, params.ttl)?;
+        sender.send((segment_key.clone(), segment_bytes))?;
```

Comment:
Can we send the Segment directly and do the serialization over on the other thread?

Reply:
[d90aaba](https://github.com/risc0/risc0/pull/2407/commits/d90aaba376ae3960dbc49a422102e0abca8dca56)